### PR TITLE
front: rollingtockeditor: UX/UI fixes

### LINKED ISSUE #4556: UX/UI improvements and fixes of the rolling stock editor
# Description

The beta version of the rolling stock editor has been released but it has a few aspects that would need to be fixed or improved.

# Goal

Improve user experience.

Unlocks a bunch of other issues :

Image

# Accepatnce criteria

- [x] change the default rolingstock image, use an existing one as a full-grey placeholder
- [x] #4564 
- [x] remove confirm modal for duplicate
- [x] add new filter for locked rollingstock
- [x] display locked icon at the left side of rollingstock-reference (same gray color)
- [x] adapt modal to the project modal style, confirm annulation by yes/no
- [x] think about the messages in modal, need smarter one for each case
- [x] bugfix several duplication (adapt new-name with date in milliseconds, the last 5 numbers) 
- [x] white background for text-input filter
- [x] fix responsive issues if any

### PR DESCRIPTION
**Pull request by Yohh**
close #4556 

- disable list while adding/updating a RS
- change RS placeholders
- change syntax of duplicated RS by adding a fragment of the date in ms
- remove modal display when duplicating a RS
- fix modal UI messages and text buttons
- display locked RS icon and add a filter for them and not locked Rs
- fix some css issues
- add and fix some translations
- auto-scroll to the RS created

### CODE REVIEW

**Comment by flomonster** [comment]
Great job for this PR!

### Small bugs / enhancements

Train id position:

image

-------

After saving the list filter is ignored (also on dev):
When we're saving rolling stock changes the list is reset like no filters are selected.

-------

Can't fill the form with floating numbers using a `.`.
In doing so, the input text box is erased.

-------

Weird spacing between curves point and the mouseover box:

image

-------

Mouse on non-clickable div:

image

**Comment by Yohh** [comment]
@flomonster 

- [x] fix header elements postions
- [x] fix filters after update
- [ ] ~~fix dot erasing input (maybe another bug issue? in fact I can't reproduce this bug)~~
- [x] fix curves pop-up
- [x] fix cursor on mouseover

**Review by flomonster — APPROVED** [review]
LGTM (only tested)

File: front/public/locales/en/rollingstock.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "addNewRollingStock": "New rolling stock",
   "cancelAction": "Would you like to cancel this action?",
+  "cancelUpdateRollingStock": "Would you like to cancel the update of this rolling stock?",
```
Comment (alexandredamiron) [inline front/public/locales/en/rollingstock.json]: "Do you really want to cancel your changes ?"

File: front/public/locales/en/rollingstock.json [hunk 2]
```diff
@@ -11,11 +12,13 @@
   },
   "compatibleVoltages": "Compatible voltages",
   "confirmAction": "Would you like to confirm this action?",
+  "confirmUpdateRollingStock": "Would you like to confirm the update of this rolling stock?",
```
Comment (alexandredamiron) [inline front/public/locales/en/rollingstock.json]: Please confirm your changes ?

File: front/public/locales/en/rollingstock.json [hunk 3]
```diff
@@ -11,11 +12,13 @@
...
   "copy": "copy",
   "curves": {
     "all": "All curves",
     "default": "Default curve"
   },
+  "deleteRollingStock": "would you like to delete this rolling stock?",
```
Comment (alexandredamiron) [inline front/public/locales/en/rollingstock.json]: Maj

File: front/public/locales/en/rollingstock.json [hunk 4]
```diff
@@ -37,7 +40,9 @@
   "inertiaCoefficient": "Factor of inertia",
   "legend": "Legend",
   "length": "Length",
+  "listDisabled": "please quit edit mode to access the list",
```
Comment (alexandredamiron) [inline front/public/locales/en/rollingstock.json]: Maj: Please

File: front/public/locales/fr/rollingstock.json [hunk 5]
```diff
@@ -38,7 +41,9 @@
   "inertiaCoefficient": "Coefficient d'inertie",
   "legend": "Légende",
   "length": "Longueur",
+  "listDisabled": "veuillez quitter l'éditeur pour modifier la séléction",
```
Comment (alexandredamiron) [inline front/public/locales/fr/rollingstock.json]: Maj

File: front/src/modules/rollingStock/components/RollingStockSelector/SearchRollingStock.tsx [hunk 6]
```diff
@@ -48,8 +51,22 @@ function rollingStockPassesSearchedStringFilter(
   ].some(includesSearchedString);
 }
```
Comment (alexandredamiron) [inline front/src/modules/rollingStock/components/RollingStockSelector/SearchRollingStock.tsx:58]: It is not obvious to read. 
please rename filterList:Filters -> filters:Filters
be more explicit about the locked param (inline doc ?)

Comment (alexandredamiron) [reply front/src/modules/rollingStock/components/RollingStockSelector/SearchRollingStock.tsx:58]: Something is either locked or unlocked. If you absolutely need to use 2 different fonctions which will cause errors, please explain why

File: front/public/locales/en/rollingstock.json (hunk 1 above)
Comment (Yohh) [reply front/public/locales/en/rollingstock.json]: done

File: front/public/locales/en/rollingstock.json (hunk 2 above)
Comment (Yohh) [reply front/public/locales/en/rollingstock.json]: done

File: front/public/locales/en/rollingstock.json (hunk 3 above)
Comment (Yohh) [reply front/public/locales/en/rollingstock.json]: done

File: front/public/locales/en/rollingstock.json (hunk 4 above)
Comment (Yohh) [reply front/public/locales/en/rollingstock.json]: done

File: front/public/locales/fr/rollingstock.json (hunk 5 above)
Comment (Yohh) [reply front/public/locales/fr/rollingstock.json]: done

File: front/src/modules/rollingStock/components/RollingStockSelector/SearchRollingStock.tsx (hunk 6 above)
Comment (Yohh) [reply front/src/modules/rollingStock/components/RollingStockSelector/SearchRollingStock.tsx:58]: done, comment added for interface

File: front/src/modules/rollingStock/components/RollingStockSelector/__tests__/RollingStockModal.spec.ts [hunk 7]
```diff
@@ -8,11 +8,15 @@ describe('rollingStockPassEnergeticModeFilters', () => {
     it('should return true (all rolling stocks are accepted)', () => {
       const filterElec = false;
       const filterThermal = false;
```
Comment (alexandredamiron) [inline front/src/modules/rollingStock/components/RollingStockSelector/__tests__/RollingStockModal.spec.ts:10]: more constistent naming please

File: front/src/modules/rollingStock/components/RollingStockSelector/SearchRollingStock.tsx [hunk 8]
```diff
@@ -75,36 +102,41 @@ type SearchRollingStockProps = {
...
     thermal: false,
+    locked: false,
+    notLocked: false,
   });
 
   const searchMateriel = (e: React.ChangeEvent<HTMLInputElement>) => {
     setFilters({ ...filters, text: e.target.value.toLowerCase() });
     if (setIsLoading) setIsLoading(true);
   };
+
```
Comment (alexandredamiron) [inline front/src/modules/rollingStock/components/RollingStockSelector/SearchRollingStock.tsx]: This is not the most stable pattern to me, it won't works if you change your target name
consider toogleFilter = (e, filterParam: string) et  onChange = {(e) => toogleFitler(e, 'locked')} when JSW call

You can keep that for refacto

File: front/src/applications/rollingStockEditor/views/RollingStockEditor.tsx [hunk 9]
```diff
@@ -110,9 +110,37 @@ export default function RollingStockEditor({ rollingStocks }: RollingStockEditor
     return listOfRollingStocks;
   }
```
Comment (alexandredamiron) [inline front/src/applications/rollingStockEditor/views/RollingStockEditor.tsx:112]: Need comments please !!!

Comment (Yohh) [reply front/src/applications/rollingStockEditor/views/RollingStockEditor.tsx:112]: done

File: front/src/modules/rollingStock/components/RollingStockSelector/SearchRollingStock.tsx (hunk 8 above)
Comment (Yohh) [reply front/src/modules/rollingStock/components/RollingStockSelector/SearchRollingStock.tsx]: done

File: front/src/modules/rollingStock/components/RollingStockSelector/__tests__/RollingStockModal.spec.ts (hunk 7 above)
Comment (Yohh) [reply front/src/modules/rollingStock/components/RollingStockSelector/__tests__/RollingStockModal.spec.ts:10]: done

**Review by alexandredamiron — APPROVED** [review]

**Pull request merged**